ExampleMod/Projectiles/ExampleLastPrismBeam.cs

```diff
@@ -0,0 +1,370 @@
+			Vector2 beamStartPos = centerFloored - Main.screenPosition;
+			Vector2 beamEndPos = beamStartPos + projectile.velocity * visualBeamLength;
+			Utils.LaserLineFraming llf = new Utils.LaserLineFraming(DelegateMethods.RainbowLaserDraw);
+
+			// Draw the outer beam.
+			// c_1 is an unnamed decompiled variable which is the render color of the beam drawn by DelegateMethods.RainbowLaserDraw.
+			Color outerBeamColor = GetBeamColor();
+			DelegateMethods.c_1 = outerBeamColor * OuterBeamOpacityMultiplier * projectile.Opacity;
+			Utils.DrawLaser(spriteBatch, tex, beamStartPos, beamEndPos, scaleVec, llf);
+
+			// Draw the inner beam, which is half size.
+			scaleVec *= 0.5f;
+			Color innerBeamColor = GetInnerBeamColor();
+			DelegateMethods.c_1 = innerBeamColor * InnerBeamOpacityMultiplier * projectile.Opacity;
+			Utils.DrawLaser(spriteBatch, tex, beamStartPos, beamEndPos, scaleVec, llf);
+			return false;
+		}
+
+		private Color GetBeamColor()
+		{
+			// This hue calculation produces a unique color for each beam based on its Beam ID.
+			// Hue ranges from 0f to 1f, both of which are pure red. This function uses modulus to range from 0.57 to 0.75,
+			// which winds up being a blue-to-purple gradient.
+			float hue = (BeamID / ExampleLastPrismHoldout.NumBeams) % 0.18f + 0.57f;
+			float saturation = 0.66f;
+			float lightness = 0.53f;
+			
+			// Main.hslToRgb converts Hue, Saturation, Lightness into a Color for general purpose use.
+			Color c = Main.hslToRgb(hue, saturation, lightness);
+			
+			// Manually reduce the opacity of the color so beams can overlap without completely overwriting each other.
+			c.A = 64;
+			return c;
+		}
+
+		// Inner beams are always pure white so that they act as a "blindingly bright" center to each laser.
+		private Color GetInnerBeamColor() => Color.White;
+
+		private void ProduceBeamDust(Color beamColor)
+		{
+			// Create one dust per frame a small distance from where the beam ends.
+			const int dustType = 15;
+			Vector2 laserEndPos = projectile.Center + projectile.velocity * (BeamLength - 14.5f * projectile.scale);
+
+			// 50% chance for the dust to come off on either side of the beam.
+			float dustAngle = projectile.rotation + (Main.rand.NextBool() ? 1f : -1f) * MathHelper.PiOver2;
+			float dustStartDist = Main.rand.NextFloat(1f, 1.8f);
+			float scale = Main.rand.NextFloat(0.7f, 1.1f);
+			Vector2 dustVel = dustAngle.ToRotationVector2() * dustStartDist;
+			Dust d = Dust.NewDustDirect(laserEndPos, 0, 0, dustType, dustVel.X, dustVel.Y, 0, beamColor, scale);
+			d.color = beamColor;
+			d.noGravity = true;
+
+			// If the beam is currently large, make the dust faster and larger to match.
+			if (projectile.scale > 1f) {
+				d.velocity *= projectile.scale;
+				d.scale *= projectile.scale;
+			}
+		}
+
+		private void ProduceWaterRipples(Vector2 beamDims)
+		{
+			WaterShaderData wsd = (WaterShaderData)Filters.Scene["WaterDistortion"].GetShader();
```

Comment:
I'd personally `var` this, but considering it's ExampleMod we don't. (Type is clear on other end of the statement)
```suggestion
			WaterShaderData shaderData = (WaterShaderData)Filters.Scene["WaterDistortion"].GetShader();
```